python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,630 @@
+            Ensure the metadata CALIBDATE fields are set to the current datetime.
+        kwargs :
+            Set of key=value pairs to assign to the metadata.
+        """
+        mdOriginal = self.getMetadata()
+        mdSupplemental = dict()
+
+        self._metadata["DETECTOR"] = self._detectorName
+        self._metadata["DETECTOR_SERIAL"] = self._detectorSerial
+
+        if setDate:
+            date = datetime.datetime.now()
+            mdSupplemental['CALIBDATE'] = date.isoformat()
+            mdSupplemental['CALIB_CREATION_DATE'] = date.date().isoformat()
+            mdSupplemental['CALIB_CREATION_TIME'] = date.time().isoformat()
+
+        for key, value in kwargs.items():
+            mdSupplemental[key] = value
+
+        mdOriginal.update(mdSupplemental)
+
+    @classmethod
+    def readText(cls, filename):
+        """Read calibration representation from a yaml/ecsv file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the calibration definition.
+
+        Returns
+        -------
+        calib : `~lsst.ip.isr.IsrCalibType`
+            Calibration class.
+        """
+        data = ''
+        try:
+            data = Table.read(filename, format='ascii.ecsv')
+        except Exception as e1:
+            try:
+                with open(filename, 'r') as f:
+                    data = yaml.load(f, Loader=yaml.CLoader)
+            except Exception as e2:
+                raise e2(f"Failed to read as ecsv as well: {e1}")
+
+        if isinstance(data, dict):
+            return cls().fromDict(data)
+        else:
+            schema = afwTable.Schema()
+            keyDict = dict()
+            for colName in data.columns:
+                try:
+                    keyDict[colName] = schema.addField(colName,
+                                                       type=data[colName].dtype.type)
+                except KeyError:
+                    keyDict[colName] = schema.addField(colName, type="String", size=32)
+
+            table = afwTable.BaseCatalog(schema)
+            table.resize(len(data))
+
+            for colName in data.columns:
+                key = keyDict[colName]
+                if key.getTypeString() == 'String':
+                    for rowNum, row in enumerate(data[colName]):
+                        table[rowNum][key] = row
+                else:
+                    table[key] = data[colName]
+
+            metadata = PropertyList()
+            for k, v in data.meta.items():
+                metadata[k] = v
+            table.setMetadata(metadata)
+
+            return cls().fromTable([table])
+
+    def writeText(self, filename):
+        """Write the calibration data to a text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
```

Comment:
I did this in Defects because Defects knew that only ecsv format was going to work. That meant I forced the suffix to be ecsv even if people tried to tell me they were writing `.txt` files.  It might be helpful for subclasses to be able to declare whether YAML or ECSV are supported and maybe have an optional param here for `format=auto|yaml|ecsv` -- forcing the suffix in yaml|ecsv mode (complaining if yaml is chosen but is not supported), allowing auto to use the supplied  suffix, and auto could even force ecsv if yaml is not supported.